hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -471,10 +469,10 @@ txOutDatum txInfo o =
 {-# INLINEABLE txOutDatum #-}
 
 hashPreSerializedCommits :: [SerializedTxOut] -> BuiltinByteString
-hashPreSerializedCommits o =
+hashPreSerializedCommits commits =
   sha2_256 . encodingToBuiltinByteString $
     encodeBeginList
-      <> foldMap (\(SerializedTxOut bytes) -> unsafeEncodeRaw bytes) o
+      <> foldMap unsafeEncodeRaw (map output $ sortBy (\a b -> compareRef (input a) (input b)) commits)
```

Comment:
Arg :'(, this is where we make the collect-com more expensive now. I don't see any way around this sorting though.. even if we provide extra information in the redeemer about the order in which inputs should be combined, it can rapidly becomes clunky.